pkg/config/model/viper.go

```diff
@@ -481,7 +481,11 @@ func (c *safeConfig) UnmarshalExact(rawVal interface{}) error {
 // ReadInConfig wraps Viper for concurrent access
 func (c *safeConfig) ReadInConfig() error {
 	c.Lock()
-	err := c.Viper.ReadInConfig()
+	err := c.configSources[SourceFile].ReadInConfig()
+	if err != nil {
+		log.Warnf("failed to read the file configuration for SourceFile Viper instance: %s", err)
```

Comment:
Is logging just enough? Should we return an error and avoid calling `c.Viper.ReadInConfig()` ?

Reply:
I really don't know the best way to do this
Maybe first read the config for the main Viper
Return err if not nil
Read the config for the Viper instance dedicated to SourceFile

Reply:
This makes sense! Working on this!